.github/workflows/desktop.yml

```diff
@@ -0,0 +1,160 @@
+name: Desktop Builds
+
+on: 
+  push:
+  pull_request:
+
+env:
+  CCACHE_DIR: ${{ github.workspace }}/ccache_dir
+
+jobs:
+  build:
+    name: ${{ matrix.os }}-${{ matrix.build_type }}-${{ matrix.architecture }}-${{ matrix.python_version }}-${{ matrix.generator }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [windows-latest, ubuntu-latest, macos-latest]
```

Comment:
is "-latest" required here?   Are we going to run on older versions that aren't latest?  If so maybe we can just tag those as older versions instead of marking all of these with "-latest".  It makes the matrix output quite verbose.

Reply:
So, github has labels for various machine configurations for their runners. I think "-latest" is on relatively more stable and tested configurations. For example, Ubuntu 18 is ubuntu-latest but they also have Ubuntu-20. I think at the very least, we should be using all the "latest" ones and maybe specifically add any newer versions if need be.

https://help.github.com/en/actions/reference/virtual-environments-for-github-hosted-runners

Reply:
Gotcha, ok, thanks.